### PR TITLE
#42 Migrate Variations Checker to reso-certification-utils

### LINKED ISSUE #42: Migrate Variations Checker to reso-certification-utils
Need to migrate this code so we can integrate into services.reso.org and the Commander.

### PR DESCRIPTION
**Pull request by darnjo**
Closes #42.

### CODE REVIEW

**Comment by darnjo** [comment]
@mohit-s96 FYI updated to use /lib instead.

File: lib/find-variations/index.js [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+                suggestion.closeMatch = true;
+              }
+
+              POSSIBLE_VARIATIONS.resources.push(suggestion);
+            }
+          }
+        });
+      } else {
+        //found standard resource - check field name variations
+        Object.keys(metadataReportMap?.[resourceName]).forEach(fieldName => {
```
Comment (mohit-s96) [inline lib/find-variations/index.js:391]: The pattern `Object.keys(metadataReportMap?.[resourceName])` would throw if `metadataReportMap?.[resourceName]` is undefined of null (`Object.keys(undefined | null)` throws). We could maybe use  `Object.keys(metadataReportMap?.[resourceName] || {})` ?

File: lib/find-variations/index.js [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+          const isStandardField = standardMetadataMap?.[resourceName]?.[fieldName] ?? false;
+
+          if (!isStandardField) {
+            //field was not found in reference metadata - look for variations
+            Object.keys(standardMetadataMap?.[resourceName]).forEach(standardFieldName => {
```
Comment (mohit-s96) [inline lib/find-variations/index.js:396]: Same as above. `Object.keys(metadataReportMap?.[resourceName] || {})` could be used instead.

File: lib/find-variations/index.js (hunk 1 above)
Comment (darnjo) [reply lib/find-variations/index.js:391]: Thanks. As a side note, have been using `??` rather than `||`.

**Review by mohit-s96 — APPROVED** [review]

**Pull request merged**